ral-facilities/operationsgateway
# DSEGOG-4 connecting API to Data tab

## Description
This allows for test data from the API to be displayed in the record table. Time to populate the `apiUrl` field in your settings file! 

The record data itself is requested from `/records` and the record count is requested from `/records/count`. To allow for the display of channel data, I'm also currently using the `/records` endpoint to gather metadata about each channel, as this currently contains a limited amount of metadata for each received channel. This request for all channel metadata is done when the site is launched, to prevent unnecessary extra requests.

Search parameters are now read from the Redux store. Changing `fromDate`, `toDate`, column `sort`, `resultsPerPage` and `page` number should result in a new query to the backend.

I've also added a basic implementation of the `Preloader` component to wait for the settings file to be read before making any backend requests. I copied it straight from DataGateway but if you're sharp you'll notice it's missing its spinning animation for some reason (not sure why but made an issue to fix it later in **DSEGOG -70**).

Near the end, I noticed a bug that was present on develop that meant the timestamp column was not fully sticky (column header was being overlapped by other column headers' menu and divider icons. Fixed this in aa2b6254244b93b5923c9fccea3f4a608b50748b

## Testing instructions
Add in the `apiUrl` provided by Matt to your settings file. Backend data should display! While searching by parameters and changing page, keep an eye on requests being made and make sure React Query is correctly caching requests.

While testing a search by date, you may notice that the way I implemented the DateTimePickers to accept a date value is pretty borked. Sometimes values are accepted and sometimes they aren't. I've made a separate issue for this in **DSEGOG -69** (I won't link it because Jira will think this PR closes it).


## Agile board tracking
Closes [DSEGOG-4](https://stfc.atlassian.net/browse/DSEGOG-4?atlOrigin=eyJpIjoiODA3OGEwMmFkMDhkNDk1MDhkYzJkZTQ5OTNhYjA1MDUiLCJwIjoiaiJ9)
